tests/gordo_components/builder/test_builder.py

```diff
@@ -50,7 +50,39 @@ def metadata_check(metadata, check_history):
     # Scores is allowed to be an empty dict. in a case where the pipeline/transformer
     # doesn't implement a .score()
     if metadata["model"]["cross-validation"]["scores"] != dict():
-        assert "explained-variance" in metadata["model"]["cross-validation"]["scores"]
+        assert [
+            f"r2-score-{tag.name.replace(' ','-')}"
+            in metadata["model"]["cross-validation"]["scores"]
+            for tag in metadata["dataset"]["tag_list"]
+        ]
+        assert "r2-score-tag-mean" in metadata["model"]["cross-validation"]["scores"]
+        assert [
+            f"explained-variance-score-{tag.name.replace(' ','-')}"
+            in metadata["model"]["cross-validation"]["scores"]
+            for tag in metadata["dataset"]["tag_list"]
+        ]
+        assert (
+            "explained-variance-score-tag-mean"
+            in metadata["model"]["cross-validation"]["scores"]
+        )
+        assert [
+            f"mean-squared-error-{tag.name.replace(' ','-')}"
+            in metadata["model"]["cross-validation"]["scores"]
+            for tag in metadata["dataset"]["tag_list"]
```

Comment:
seems like lots of `metadata["dataset"]["tag_list"]` followed by `tag.name.replace(...)` maybe at the start of this conditional, do the preprocessing into `tag_list` and probably `scores` from `metadata["model"]["cross-validation"]["scores"]` 

That might actually save you a bunch of lines from the formatting.